05.wc/wc.rb

```diff
@@ -0,0 +1,70 @@
+# ! /usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+STDIN_PATTERN = /^-$/
+TOTAL = '合計'
+
+def main
+  options = parse_options(ARGV)
+  rows = build_rows(options)
+  show_rows(rows)
+end
+
+def build_rows(options)
+  rows = []
+  ARGF.each(nil) do |input_text|
+    counts = {}
+
+    counts[:filename] = ARGF.filename
+    counts[:line] = input_text.lines.count if options[:l]
+    counts[:word] = input_text.split(/\s+/).size if options[:w]
+    counts[:byte] = input_text.size if options[:c]
+    rows << counts
+  end
```

Comment:
```suggestion
  rows = ARGF.each(nil).map do |input_text|
    counts = { filename: ARGF.filename }
    counts[:line] = input_text.lines.count if options[:l]
    counts[:word] = input_text.split(/\s+/).size if options[:w]
    counts[:byte] = input_text.size if options[:c]
    counts
  end
```

空の配列を用意してループで詰め込むだけの処理はmapで置き換えられます。（チェリー本参照）

あとfilenameは最初から入れてしまうのもアリです